cmd/kubermatic-installer/cmd_quickstart.go

```diff
@@ -0,0 +1,377 @@
+	unstructured.SetNestedField(uk.Object, kkpEndpoint, "spec", "ingress", "domain")
+	unstructured.SetNestedField(uk.Object, nil, "spec", "ingress", "certificateIssuer")
+	unstructured.SetNestedField(uk.Object, fmt.Sprintf("http://%v/dex", kkpEndpoint), "spec", "auth", "tokenIssuer")
+	unstructured.SetNestedField(uk.Object, randomString(32), "spec", "auth", "issuerClientSecret")
+	unstructured.SetNestedField(uk.Object, randomString(32), "spec", "auth", "issuerCookieKey")
+	unstructured.SetNestedField(uk.Object, randomString(32), "spec", "auth", "serviceAccountKey")
+	// TODO: backport https://github.com/kubermatic/dashboard/pull/5867 or wait for new release?
+	unstructured.SetNestedField(uk.Object, "wozniakjan/dashboard", "spec", "ui", "dockerRepository")
+	unstructured.SetNestedField(uk.Object, "latest", "spec", "ui", "dockerTag")
+	kout, err := yaml.Marshal(uk.Object)
+	if err != nil {
+		logger.Fatalf("failed to marshal kubermatic.yaml: %v", err)
+	}
+	if err := os.WriteFile(kubermaticPath, kout, 0600); err != nil {
+		logger.Fatalf("failed to create kubermatic.yaml: %v", err)
+	}
+
+	vc, err := os.ReadFile(valuesExamplePath)
+	if err != nil {
+		logger.Fatalf("failed to read values.yaml example: %v", err)
+	}
+	uv := make(map[any]any)
+	if err := yaml.Unmarshal(vc, uv); err != nil {
+		logger.Fatalf("failed to decode example values.yaml: %v", err)
+	}
+
+	setNestedField(uv, "http", "dex", "ingress", "scheme")
+	setNestedField(uv, kkpEndpoint, "dex", "ingress", "host")
+	clients := uv["dex"].(map[any]any)["clients"].([]any)
+	for i, _ := range clients {
+		clientsMap := clients[i].(map[any]any)
+		setNestedField(clientsMap, randomString(32), "secret")
+		uris := clientsMap["RedirectURIs"].([]any)
+		for uri, _ := range uris {
+			u, err := url.Parse(uris[uri].(string))
+			if err != nil {
+				logger.Fatalf("failed to modify values.yaml: %v", err)
+			}
+			u.Scheme = "http"
+			u.Host = kkpEndpoint
+			uris[uri] = u.String()
+		}
+	}
+
+	setNestedField(uv, uuid.NewString(), "telemetry", "uuid")
+	setNestedField(uv, nil, "minio")
+	vout, err := yaml.Marshal(uv)
+	if err != nil {
+		logger.Fatalf("failed to marshal values.yaml: %v", err)
+	}
+	if err := os.WriteFile(valuesPath, vout, 0600); err != nil {
+		logger.Fatalf("failed to create values.yaml: %v", err)
+	}
+
+	ensureResource(kubeClient, logger, &kindIngressControllerNamespace)
+	ensureResource(kubeClient, logger, &kindKubermaticNamespace)
+	ensureResource(kubeClient, logger, &kindStorageClass)
+	ensureResource(kubeClient, logger, &kindIngressControllerService)
+	ensureResource(kubeClient, logger, &kindNodeportProxyService)
+
+	// TODO: use cmd_deploy.go instead
+	kubeconfig := filepath.Join(dir, "kube-config.yaml")
+	executable, err := os.Executable()
+	if err != nil {
+		logger.Fatal(err)
+	}
+	cmd := exec.Command(executable, "deploy", "--config", kubermaticPath, "--helm-values", valuesPath, "--kubeconfig", kubeconfig)
```

Comment:
just for simplification, this calls the `kubermatic-installer` again but would be best to refactor to use the code from there instead.

Reply:
I would like to leave this as todo for followup task

Reply:
I don't want to push on that too much, but is there any problem with calling this? https://github.com/kubermatic/kubermatic/blob/84791456e555486fe0194776a63682154908cc33/pkg/install/stack/kubermatic-master/stack.go#L85

It seems pretty equal in complexity.

Reply:
that is a good push actually, seems like it wasn't as hard as I was worried - https://github.com/kubermatic/kubermatic/pull/12216/commits/8c995e29c19c6614f9478500812de9b261629b99

the only reason I did it via calling itself is that I had this all initially as a bash script (which was supposed to be used in `curl | bash` fashion. Later decided to integrate it into the installer instead.